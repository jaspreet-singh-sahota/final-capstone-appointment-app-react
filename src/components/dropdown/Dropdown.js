import React from 'react';
import PropTypes from 'prop-types';
import cities from '../../redux/citiesData';
import StyledSelect from './styles/dropdown';
import './styles/dropdown.css';

const Dropdown = ({ setCity }) => {
  const onChange = e => {
    setCity(e[0].value);
  };

  return (
    <div>
      <StyledSelect style={{ textAlign: 'center', marginLeft: '20px' }} options={cities} onChange={value => onChange(value)} required placeholder="--Select City--" />
    </div>
  );
};

Dropdown.propTypes = {
  setCity: PropTypes.func,
};

Dropdown.defaultProps = {
  setCity: undefined,
};

export default Dropdown;
