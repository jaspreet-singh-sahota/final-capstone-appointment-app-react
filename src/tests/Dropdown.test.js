import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from '../components/dropdown/Dropdown';
import StyledSelect from '../components/dropdown/styles/dropdown';

describe('Dropdown', () => {
  it('should render a <StyledSelect /> component', () => {
    const wrapper = shallow(<Dropdown />);
    expect(wrapper.find(StyledSelect)).toHaveLength(1);
  });

  it('matches Render snapshot', () => {
    const wrapper = shallow(<Dropdown />);
    expect(wrapper).toMatchSnapshot();
  });
})

