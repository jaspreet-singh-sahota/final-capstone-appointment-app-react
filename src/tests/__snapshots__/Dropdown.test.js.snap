// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown matches Render snapshot 1`] = `ShallowWrapper {}`;
